app/src/ui/repository.tsx

```diff
@@ -545,4 +549,27 @@ export class RepositoryView extends React.Component<
     }
     return null
   }
+
+  /**
+   * This method is a generic event handler for when a commit has ended being
+   * dragged.
+   *
+   * Currently only used for cherry picking, but this could be more generic.
+   */
+  private onDragCommitEnd = async () => {
+    this.props.dispatcher.closeFoldout(FoldoutType.Branch)
+
+    // On drag end fires before the drop event, we need to wait here and see if
+    // the drop event results in a change of cherry picking state.
+    await sleep(10)
```

Comment:
This is very scary IMO. If it takes any longer to update the state this logic would break, no?

Can't we listen to the drop event instead?

Reply:
@tidy-dev and I discussed about this. We need a slightly bigger refactor to get rid of this `sleep`, so we're gonna go ahead and merge this PR, and then tackle that problem separately and ASAP!!